### PR TITLE
feat: `File` - Milestone 3

### LINKED ISSUE #2914: File: Milestone 3
Add something like the following:

```
def mkdir(d: String): Result[Bool, String]
def mkdirs(d: String): Result[Bool, String]
```

Write some comments so we can agree what each should do.

### PR DESCRIPTION
**Pull request by ninaandrup**
Closes #2914

### CODE REVIEW

**Comment by ninaandrup** [comment]
@magnus-madsen Do you agree with the comments?

**Comment by magnus-madsen** [comment]
I prose it should say: Returns true if the directory was created and did not already exist. 

Thus:

Ok(true) - Created and did not exist.
Ok(false) - Directory already exists (no problem).
Error(msg) - Something went bad.

**Comment by ninaandrup** [comment]
> I prose it should say: Returns true if the directory was created and did not already exist.
> 
> Thus:
> 
> Ok(true) - Created and did not exist. Ok(false) - Directory already exists (no problem). Error(msg) - Something went bad.

Ah, yes, that is a more clear description

File: main/src/library/File.flix [hunk 1]
```diff
@@ -800,4 +800,19 @@ namespace File {
...
+    /// Creates the directory `d`.
+    ///
+    /// Returns `true` if the directory `d` was created and did not already exist.
+    ///
+    pub def mkdir(d: String): Result[Bool, String] = ???
+
+    ///
+    /// Creates the directory `d` along with all necessary parent directories.
+    ///
+    /// Returns `true` if the directory `d` was created and did not already exist.
```
Comment (ninaandrup) [inline main/src/library/File.flix:813]: Is this okay?
I split the description up to make it shorter. It is more like the `def write*`'s now.

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**